File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
         int left = 0;
        int right = numbers.size() - 1; // Use .size() instead of .length()
        vector<int> result(2); // Vector to store the indices

        while (left < right) {
            int sum = numbers[left] + numbers[right];

            if (sum == target) {
                result[0] = left + 1;  
                result[1] = right + 1; 
                return result;         
            } else if (sum > target) {
                right--;
            } else {
                left++;
            }
        }
        return result;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        n = len(numbers)

        for i in range(n):
            comp = target -numbers[i]
            low = i+1
            high =n-1

            while low<=high:
                mid = (low+high)//2

                if(numbers[mid]==comp):
                    return [i+1,mid+1]
                elif (numbers[mid]<comp):
                    low=mid+1
                else:
                    high =mid-1
        return []


            